template/src/context/AuthContext.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import cache from '../utils/cache';
+
+type AuthContextData = {
+  isAuthenticated: boolean;
+  authenticate: ({
+    token,
+    persist,
+  }: {
+    token: string;
+    persist?: boolean;
+  }) => void;
+  logout: () => void;
+};
+
+type ProviderProps = {
+  children: React.ReactNode;
+};
+
+export const AuthContext = React.createContext<AuthContextData>(
+  {} as AuthContextData
+);
+
+export function AuthProvider({ children }: ProviderProps) {
+  const { setCache } = cache;
+  const [isAuthenticated, setIsAuthenticated] = React.useState(false);
+
+  function authenticate({
+    token,
+    persist,
+  }: {
+    token: string;
+    persist?: boolean;
+  }) {
+    setCache(token);
```

Comment:
Why set the `token` variable? You are already using [`localStorage`](https://github.com/t10d/cra-template-kingdom/blob/56d775475e4cdceaa617c0182f2d1557fb8c7a9e/template/src/context/AuthContext.tsx#L38) to store the `token`. That is why I see this [`setCache` implementation](https://github.com/t10d/cra-template-kingdom/blob/56d775475e4cdceaa617c0182f2d1557fb8c7a9e/template/src/utils/cache.ts#L4) unnecessary.